upstairs/src/downstairs.rs

```diff
@@ -558,14 +566,52 @@ impl Downstairs {
 
         // Fire DTrace probes and update stats
         let io_size = done.io_size();
-        match &done.work {
+        let work = done.work.clone();
+        Self::update_io_done_stats(&self.stats, work, ds_id, io_size);
```

Comment:
Ahh. github diff made a mess of this.

I pulled the stat update part of `ack_job()` out to a stand alone function, but
the diff thinks I changed more of `ack_job()` than I really did.  There should be
almost nothing else different